haxe/ui/macros/ComponentMacros.hx

```diff
@@ -103,113 +102,112 @@ class ComponentMacros {
         var c:ComponentInfo = ComponentParser.get(MacroHelpers.extension(f)).parse(File.getContent(f), new FileResourceResolver(f));
         //trace(c);
 
-        var code:String = "function() {\n";
         for (styleString in c.styles) {
-            code += "haxe.ui.Toolkit.styleSheet.addRules('" + styleString + "');\n";
+            code.push(macro haxe.ui.Toolkit.styleSheet.addRules($v{styleString}));
         }
 
-        code += buildComponentCode(c, 0, namedComponents);
-        code += assignBindings(c.bindings);
+        buildComponentCode(code, c, 0, namedComponents);
+        assignBindings(code, c.bindings);
 
         var fullScript = "";
         for (scriptString in c.scriptlets) {
             fullScript += scriptString;
         }
-        code += "c0.script = '" + fullScript + "';\n";
-
-        code += "return c0;\n";
-        code += "}()\n";
+        code.push(macro c0.script = $v{fullScript});
+        code.push(macro c0);
 
-        return code;
+        return macro @:pos(Context.currentPos()) $b{code};
```

Comment:
We're no longer generating the `(function() code)()` wrapping because it shouldn't be necessary.